init.go

```diff
@@ -225,18 +215,17 @@ func (a *initFnCmd) buildFuncFile(c *cli.Context) error {
 		return errors.New("function file runtime is 'docker', but no Dockerfile exists")
 	}
 
-	var rt string
+	var helper langs.LangHelper
 	if a.Runtime == "" {
-		rt, err = detectRuntime(wd)
+		helper, err = detectRuntime(wd)
 		if err != nil {
 			return err
 		}
-		a.Runtime = rt
-		fmt.Printf("Found %v function, assuming %v runtime.\n", rt, rt)
+		fmt.Printf("Found %v function, assuming %v runtime.\n", helper.Runtime(), helper.Runtime())
 	} else {
 		fmt.Println("Runtime:", a.Runtime)
+		helper = langs.GetLangHelper(a.Runtime)
 	}
-	helper := langs.GetLangHelper(a.Runtime)
 	if helper == nil {
 		fmt.Printf("Init does not support the %s runtime, you'll have to create your own Dockerfile for this function", a.Runtime)
```

Comment:
Not direclty related to this change but setting an unknown runtime gives a garbled error due to missing newline here: 
```
OCliffe@OCLIFFE-mac /t/foojazz> fn init --runtime asdas
Runtime: asdas
Init does not support the asdas runtime, you'll have to create your own Dockerfile for this functionERROR: could not detect entrypoint or cmd for asdas, use --entrypoint and/or --cmd to set them explicitly
```

Reply:
fixed.